pymodbus/datastore/simulator.py

```diff
@@ -762,7 +762,7 @@ def validate_type(self, func_code, real_address, count):
             check = (CellType.UINT16, CellType.STRING)
             reg_step = 1
         else:
-            check = (CellType.UINT32, CellType.FLOAT32, CellType.STRING)
+            check = (CellType.UINT32, CellType.FLOAT32, CellType.STRING)  # type: ignore[assignment]
```

Comment:
That sounds wrong...that you need to ignore the assignment. Line 762 is ok....check contains the legal CellTypes and can be any number (of the valid types).

Reply:
`mypy` does not like the fact that one path has a `tuple` with 2 items, and another with 3.

I don't see any meaningful consequence to the warning, so I ignored it.

Reply:
Do you prefer the other alternative of annotating `check: tuple` before any of the paths define it?

Reply:
> sadly enough it is not possible to annotate without assigning

Why do you say that?  It's perfectly possible with no assignment or `Non`

Reply:
ok, lets see what you can come up with....I thought annotation was only possible with assigning.